### PR TITLE
Added tests for listing configmap for all roles

### PR DESCRIPTION
**Pull request by soumyalj**
Please review

### CODE REVIEW

File: tests/validation/tests/v3_api/test_configmaps.py [hunk 1]
```diff
@@ -167,6 +186,26 @@ def test_rbac_cmap_cluster_member_edit(remove_resource):
...
 
 
+@if_test_rbac
+def test_rbac_cmap_cluster_member_list(remove_resource):
+
+    user_token = rbac_get_user_token_by_role(CLUSTER_MEMBER)
+    project, ns = create_project_and_ns(user_token, namespace["cluster"],
+                                        random_test_name("rbac-cluster-mem"))
+    p_client = get_project_client_for_token(project, user_token)
+    rbac_configmap_create(p_client, ns)
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_configmaps.py:196]: what is the purpose of this line?

File: tests/validation/tests/v3_api/test_configmaps.py [hunk 2]
```diff
@@ -167,6 +186,26 @@ def test_rbac_cmap_cluster_member_edit(remove_resource):
...
+    configmap = rbac_configmap_create(p_client, ns)
+    cmapname = configmap["name"]
+    print(cmapname)
+    cmapdict = p_client.list_configMap(name=cmapname)
+    print(cmapdict)
+    assert cmapdict.resourceType == "configMap"
+    configmapdata = cmapdict.get('data')
+    assert len(configmapdata) == 1
+    assert configmapdata[0].name == cmapname
+    remove_resource(configmap)
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_configmaps.py:206]: Should the project also be removed?

Comment (soumyalj) [reply tests/validation/tests/v3_api/test_configmaps.py:206]: Removed it

File: tests/validation/tests/v3_api/test_configmaps.py (hunk 1 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_configmaps.py]: Removed the redundant line

File: tests/validation/tests/v3_api/test_configmaps.py (hunk 2 above)
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_configmaps.py:206]: Removed the project

**Comment by sangeethah** [comment]
@soumyalj We should be able to use @pytest.mark.parametrize for the new cases added .

**Comment by soumyalj** [comment]
@sangeethah : Parametrized create, edit, delete, list config map tests.

File: tests/validation/tests/v3_api/test_configmaps.py [hunk 3]
```diff
@@ -329,6 +331,8 @@ def rbac_configmap_create(p_client, ns):
     create_and_validate_workload_with_configmap_as_volume(p_client, configmap,
                                                           ns, keyvaluepair)
 
+    return configmap
+
 
 def rbac_configmap_edit(p_client, ns):
```
[the remark this answers is not in the document]
Comment (soumyalj) [reply tests/validation/tests/v3_api/test_configmaps.py:337]: Deleted the config map in rbac_configmap_edit()

**Pull request merged**